Sources/App/Core/SignificantBuilds.swift

```diff
@@ -0,0 +1,112 @@
+        }
+    }
+
+    var builds: [BuildInfo]
+
+    // TODO: add tests
+    static func query(on database: Database, owner: String, repository: String) -> EventLoopFuture<Self> {
+        return Build.query(on: database)
+            .join(parent: \.$version)
+            .join(Package.self, on: \App.Version.$package.$id == \Package.$id)
+            .join(Repository.self, on: \Repository.$package.$id == \Package.$id)
+            .filter(App.Version.self, \App.Version.$latest != nil)
+            .filter(Repository.self, \.$owner, .custom("ilike"), owner)
+            .filter(Repository.self, \.$name, .custom("ilike"), repository)
+            .field(\.$platform)
+            .field(\.$swiftVersion)
+            .field(\.$status)
+            .all()
+            .mapEach {
+                BuildInfo($0.swiftVersion, $0.platform, $0.status)
+            }
+            .map(Self.init(builds:))
+    }
+}
+
+
+extension SignificantBuilds {
+    enum CompatibilityResult<Value: Equatable>: Equatable {
+        case available([Value])
+        case pending
+
+        var values: [Value]? {
+            switch self {
+                case .available(let values):
+                    return values
+                case .pending:
+                    return nil
+            }
+        }
+    }
+
+    /// Returns platform compatibility across a package's significant versions.
+    /// - Returns: A `CompatibilityResult` of `Platform`
+    func platformCompatibility() -> CompatibilityResult<Build.Platform> {
+        if builds.allSatisfy({ $0.status == .triggered }) { return .pending }
+
+        let builds = builds
+            .filter { $0.status == .ok }
+        let compatibility = Build.Platform.allActive.map { platform -> (Build.Platform, Bool) in
+            for build in builds {
+                if build.platform == platform {
+                    return (platform, true)
+                }
+            }
+            return (platform, false)
+        }
+        return .available(
+            compatibility
+                .filter { $0.1 }
+                .map { $0.0 }
+        )
+    }
+
+    /// Returns swift versions compatibility across a package's significant versions.
+    /// - Returns: A `CompatibilityResult` of `SwiftVersion`
+    func swiftVersionCompatibility() -> CompatibilityResult<SwiftVersion> {
+        if builds.allSatisfy({ $0.status == .triggered }) { return .pending }
+
+        let builds = builds
+            .filter { $0.status == .ok }
+        let compatibility = SwiftVersion.allActive.map { swiftVersion -> (SwiftVersion, Bool) in
+            for build in builds {
+                if build.swiftVersion.isCompatible(with: swiftVersion) {
+                    return (swiftVersion, true)
+                }
+            }
+            return (swiftVersion, false)
+        }
+        return .available(
+            compatibility
+                .filter { $0.1 }
+                .map { $0.0 }
+        )
+    }
+}
```

Comment:
The functions in this extension used to live in `PackageResult`, they're not new.